ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala

```diff
@@ -210,21 +211,26 @@ private[testtool] final class ParticipantTestContext private[participant] (
   def waitForParties(
       otherParticipants: Iterable[ParticipantTestContext],
       expectedParties: Set[Party],
-  ): Future[Unit] = eventually {
-    val participants = otherParticipants.toSet + this
-    Future
-      .sequence(participants.map(otherParticipant => {
-        otherParticipant
-          .listParties()
-          .map(actualParties => {
-            assert(
-              expectedParties.subsetOf(actualParties),
-              s"Parties from $this never appeared on $otherParticipant.",
-            )
-          })
-      }))
-      .map(_ => ())
-  }
+  ): Future[Unit] =
+    if (waitForPartiesEnabled) {
+      eventually {
+        val participants = otherParticipants.toSet + this
+        Future
+          .sequence(participants.map(otherParticipant => {
```

Comment:
As you are just asserting that the parties have been allocated, it is enough to use `foreach` on the created futures instead of `map`. I.e.:
```scala
Future
  .sequence(participants.map { otherParticipant =>
    otherParticipant
      .listParties()
      .foreach(/* assert here */)
      ...
```

Reply:
:+1:

Reply:
Hm, I don't see how a foreach works here. Its type is Unit and we need a Future here for this to work with the eventually combinator. I'll leave it as is.

Reply:
Sorry, my oversight -- you're right, because of the `eventually` combinator you need to use `.map`.